apps/studio/pages/project/[ref]/observability/database.tsx

```diff
@@ -140,38 +140,43 @@ const DatabaseUsage = () => {
     },
   })
 
-  const onRefreshReport = async () => {
-    if (!selectedDateRange) return
+  const onRefreshReport = createRefreshHandler(
+    datePickerValue,
+    datePickerHelpers,
+    handleDatePickerChange,
+    async () => {
+      if (!selectedDateRange) return
 
-    setIsRefreshing(true)
-    refresh()
-    const { period_start, period_end, interval } = selectedDateRange
-    REPORT_ATTRIBUTES.forEach((chart: any) => {
-      chart.attributes.forEach((attr: any) => {
+      setIsRefreshing(true)
+      refresh()
+      const { period_start, period_end, interval } = selectedDateRange
+      REPORT_ATTRIBUTES.forEach((chart: any) => {
+        chart.attributes.forEach((attr: any) => {
+          queryClient.invalidateQueries({
+            queryKey: analyticsKeys.infraMonitoring(ref, {
+              attribute: attr.attribute,
+              startDate: period_start.date,
+              endDate: period_end.date,
+              interval,
+              databaseIdentifier: state.selectedDatabaseId,
+            }),
+          })
+        })
+      })
```

Comment:
Maybe flat map canidate? 



```suggestion
REPORT_ATTRIBUTES
  .flatMap(chart => chart.attributes)
  .forEach(attr => {
    queryClient.invalidateQueries({
      queryKey: analyticsKeys.infraMonitoring(ref, {
        attribute: attr.attribute,
        startDate: period_start.date,
        endDate: period_end.date,
        interval,
        databaseIdentifier: state.selectedDatabaseId,
      }),
    })
  })

```